<launch>
  <!--img capture node-->
  <node name="ImgPublisher" pkg="img_capture" type="imgCapture" output="screen" required="true">
    <param name="img_quality" value="100" />
    <param name="system_rate" value="60" />
    <param name="network_smooth" value="true" />
    <param name="cam_id" value="1" />
  </node>

  <!--apriltag node-->
  <node name="ApriltagHandler" pkg="img_capture" type="apriltagFinder" output="screen" >
    <param name="camera_info_url" value="package://img_capture/cameraInfo/camera.yaml"/>
    <param name="default_tag_size" value="0.037" />
    <param name="viewer" value="true" />
    <param name="publish_detections_image" value="false" />
    <param name="display_type" value="CUBE" />
    <param name="marker_thickness" value="0.02" />
    <rosparam command="delete" param="tag_data" />
   <param name="imgSource" value="ImgPublisher" />
    <rosparam param="tag_data">
      "1": 
        size: 0.035
      "2":
        size: 0.048
    </rosparam>
  </node>

  <!--img viewer-->
  <node name="ImgViewer" pkg="img_capture" type="imgSubscriberTest" output="screen">
    <param name="imgSource" value="ImgPublisher" />
  </node>

  <!--apriltag infos sync node-->
  <node name="Synchronizer" pkg="img_capture" type="synchronizer" output="screen" required="true">
    <param name="cam_num" value="1" />
    <param name="info_prefix" value="/apriltag_info" />
  </node>	

</launch>
